### PR TITLE
udev: fix race in renaming network interface

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/udev-event.c [hunk 1]
```diff
@@ -881,8 +881,7 @@ static int update_devnode(UdevEvent *event) {
                 return log_device_error_errno(dev, r, "Failed to get devnum: %m");
 
         /* remove/update possible left-over symlinks from old database entry */
-        if (event->dev_db_clone)
-                (void) udev_node_update_old_links(dev, event->dev_db_clone);
+        (void) udev_node_update_old_links(dev, event->dev_db_clone);
```
Comment (keszybz) [inline src/udev/udev-event.c:884]: Hmm, why is this check unnecessary? udev_node_update_old_links() needs the args to be nonnull, and I don't see how this is guaranteed here.

File: src/udev/udev-event.c [hunk 2]
```diff
@@ -945,8 +944,7 @@ static int udev_event_on_move(UdevEvent *event) {
         sd_device *dev = event->dev;
         int r;
 
-        if (event->dev_db_clone &&
-            sd_device_get_devnum(dev, NULL) < 0) {
+        if (sd_device_get_devnum(dev, NULL) < 0) {
                 r = device_copy_properties(dev, event->dev_db_clone);
```
Comment (keszybz) [inline src/udev/udev-event.c:948]: Same here, how do we konw dev_db_clone is nonnull?

File: src/udev/udev-event.c (hunk 1 above)
Comment (yuwata) [reply src/udev/udev-event.c:884]: `dev_db_clone` is always created except for 'remove' uevent. `update_devnode()` is not called on 'remove' uevent. So, it is not necessary.

File: src/udev/udev-event.c (hunk 2 above)
Comment (yuwata) [reply src/udev/udev-event.c:948]: On 'move' uevent, `dev_db_clone` is always created.

Comment (keszybz) [reply src/udev/udev-event.c:948]: We'll get an assert failure is this is not satisfied. Should be OK to merge.

**Pull request merged**